sdk/maps/test-resources.json

```diff
@@ -0,0 +1,99 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "baseName": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The base resource name."
+      }
+    },
+    "tenantId": {
+      "type": "string",
+      "defaultValue": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+      "metadata": {
+        "description": "The tenant ID to which the application and resources belong."
+      }
+    },
+    "testApplicationOid": {
+      "type": "string",
+      "metadata": {
+        "description": "The client OID to grant access to test resources."
+      }
+    },
+    "testApplicationId": {
+      "type": "string",
+      "metadata": {
+        "description": "The application client id used to run tests."
+      }
+    },
+    "testApplicationSecret": {
+      "type": "string",
+      "metadata": {
+        "description": "The application client secret used to run tests."
+      }
+    }
+  },
+  "variables": {
+    "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/8f5e0ce6-4f7b-4dcf-bddf-e6f48634a204')]",
+    "mapsAccountName": "[guid(resourceGroup().id, deployment().name, parameters('baseName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Maps/accounts",
+      "apiVersion": "2021-12-01-preview",
+      "name": "[variables('mapsAccountName')]",
+      "location": "eastus",
+      "sku": {
+        "name": "G2",
+        "tier": "Standard"
+      },
+      "kind": "Gen2",
+      "identity": {
+        "type": "None"
+      },
+      "properties": {
+        "disableLocalAuth": false,
+        "cors": {
+          "corsRules": [
+            {
+              "allowedOrigins": ["*"]
+            }
+          ]
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2018-09-01-preview",
+      "name": "[guid(resourceGroup().id)]",
+      "properties": {
+        "roleDefinitionId": "[variables('roleDefinitionId')]",
+        "principalId": "[parameters('testApplicationOid')]"
+      }
+    }
+  ],
+  "outputs": {
+    "AZURE_TENANT_ID": {
+      "type": "string",
+      "value": "[parameters('tenantId')]"
+    },
+    "AZURE_CLIENT_ID": {
+      "type": "string",
+      "value": "[parameters('testApplicationId')]"
+    },
+    "AZURE_CLIENT_SECRET": {
+      "type": "string",
+      "value": "[parameters('testApplicationSecret')]"
+    },
```

Comment:
Please remove these (and any corresponding parameters that are unused). You can reference them in test via the `MAPS` prefix instead of `AZURE`.

Reply:
Is that the standard you want us to follow @benbp?

We have our karma configs, environment variables and recording configs, fake envs to mask the actual values pointing to AZURE_CLIENT_SECRET, AZURE_CLIENT_ID and AZURE_TENANT_ID values in almost all the SDKs.

Reply:
I think those variables are used by the [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/identity/identity#environment-variables) . We only use the subscription key for test right now, but I think we can keep this if we need to test with the Azure AD?

Reply:
Oh yes, that too.
Our identity credentials assume these variable names.

Reply:
@HarshaNalluru historically we have namespaced the variables with the service directory name so as not to break multi-tenant/multi-auth testing scenarios.

The workaround for this in the pipeline is to set the `AZURE_*` variables in the yaml similar to: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/personalizer/ai-personalizer-rest/tests.yml . Please do it in the yaml and not in the ARM template, because any secret set as ARM deployment outputs is available to anyone in the subscription with reader access (we are working on better handling of our credentials).

However, this is still annoying because if you want to do local testing, the tests will more likely end up using your own credentials via `DefaultAzureCredential` rather than the test service principal provisioned by the deployment script. In some of the other languages there are common test helpers that look for the service directory prefixed variables first when building the client.

I think this is confusing enough for people that I will try to take another pass at fixing it, but in the meantime, I recommend following the example I posted above.

Reply:
Ok, thanks, that makes sense.
Now I understand why I keep having to change the prefixes of the env vars generated using test-resources.json in my local development. 😎